lib/shopify_app/login_protection.rb

```diff
@@ -27,7 +27,7 @@ def shop_session
     end
 
     def login_again_if_different_shop
-      if shop_session && params[:shop] && params[:shop].is_a?(String) && shop_session.url != params[:shop]
+      if shop_session && params[:shop] && (shop_session.url != params[:shop])
```

Comment:
:ok: 

+1 for reducing the public interface. There is also some weird abstraction lines between this and some of the other concerns for legacy reasons.